### DISCUSSION TITLE
I2C usage

### DISCUSSION THREAD

**Discussion by dwarning** [Rodemfr/MicroNav#4] [Technical discussions]
Not a issue - perhaps better an discussion topic. Can you open Discussion?
I want ask you how you plan to connect display and DLHC together to I2C bus and solve the assembly case. Do you soldered a kind of adapter to the t-beam board to contact the bus?

**Comment by Rodemfr** [comment]
No plan so far. I did not connect yet my LSM303 on my T-Beam.
I saw on internet some [3D printed boxes for T-Beam](https://cults3d.com/en/3d-model/tool/ttgo-t-beam-v1-1-case-for-big-gps-antenna) with extra space to use a big GPS antenna instead of the small default one. We could imagine doing the same for LSM303 : a 3D printed box with some extra space to put a LSM303DLHC breakout board and then just solder a few wires to the same I2C bus than display ?

**Reply by frgaude** [reply]
I was thinking on this one that the bigger gps antenna might be needed for a setup inside the boat.

**Comment by frgaude** [comment]
I came by this interesting commit on the lilygo github page that suggests they might be adding a few extra chips on the next iteration of the t-beam: compass, accelerometer and rtc.
This board will be based on an ESP32 S3.
No details on availability yet. 

https://github.com/Xinyuan-LilyGO/LilyGo-LoRa-Series/commit/eb17152739c27ab4d71333fda87601ac544456fb

**Comment by frgaude** [comment]
On my side I mounted the OLED screen on a female pin header. So I can swap it with the LSM303, but that's admittedly not a real solution. 
It does rise the OLED screen enough so that I can still access the other GPIOs and maybe mount the LSM303 beneath but that might be far from an ideal placement for it.

**Comment by Rodemfr** [comment]
A new version of T-Beam with a compass or IMU would fit exactly the need for MicroNav ! It would make a terrific project with no soldering at all. Already the T-Beam V1.1 is impressive : I added the possibility to switch between various status and configuration pages with the display and press button and everything can now be configured without recompiling or connecting a terminal. That's really convenient. Everybody will have the same HW and the same software, it will ease testing and bugfixing !
I will check what are the differences between ESP32 S3 and the current one...
